NewsBook/Controllers/UsersController.cs

```diff
@@ -19,105 +18,91 @@ namespace NewsBook.Controllers
         public async Task<IActionResult> Get(
             [FromQuery] bool usePaging, 
             [FromQuery] PagingParameters pagingParameters
-            )
+        )
         {
             if(usePaging == true)
             {
-                var pagedUser = await _usersRepository.GetAll(pagingParameters);
-                var pagedUsersDTO = new PagedList<UserReadDTO>
-                {
-                    Items = _mapper.Map<List<UserReadDTO>>(pagedUser.Items),
-                    TotalCount = pagedUser.TotalCount,
-                    TotalPages = pagedUser.TotalPages,
-                    CurrentPage = pagedUser.CurrentPage,
-                    PageSize = pagedUser.PageSize
-                };
-                return Ok(pagedUsersDTO);
+                var paginatedUsers = await _mediator.Send(new GetPaginatedUsersQuery { 
+                    Page = pagingParameters
+                });
+                return Ok(paginatedUsers);
             }
-            var user = await _usersRepository.GetAll();
+
+            var user = await _mediator.Send(new GetUsersQuery());
             return Ok(_mapper.Map<List<UserReadDTO>>(user));
         }
 
         [Authorize(Role.admin)]
         [HttpGet("{id}")]
         public async Task<IActionResult> Get(Guid id)
         {
-            return Ok(await _usersRepository.GetById(id));
+            return Ok(await _mediator.Send(new GetUserByIdQuery { Id = id }));
         }
+
         [AllowAnonymous]
         [HttpPost]
         public async Task<IActionResult> Post([FromBody] UserWriteDTO userDTO)
         {
-            var email = await _dbContext.Users.FirstOrDefaultAsync(value => value.Email.Equals(userDTO.Email));
-            if(email == null)
+            var user = await _mediator.Send(new PostUserQuery
             {
-                var user = await _usersRepository.Insert(userDTO.Name, userDTO.Email, userDTO.Password);
-                return Ok(_mapper.Map<UserReadDTO>(user));
-            }
-            return BadRequest("Email already Exist");
+                userDTO = userDTO
+            });
+            return Ok(user);
         }
+
         [AllowAnonymous]
         [HttpPut]
         public async Task<IActionResult> Put(
             [FromBody] UserWriteDTO userDTO
             )
         {
-            var userId = _identityServices.GetUserId() ?? Guid.Empty;
-            if (userId.Equals(_usersRepository.GetById(userId))) 
+            var user = await _mediator.Send(new PostUserQuery
             {
-                await _usersRepository.Insert(userDTO.Name, userDTO.Email, userDTO.Password);
-                return Ok(userId);
-            }
-            return BadRequest("User not found");
+                userDTO = userDTO
+            });
+            return Ok(user);    
         }
-        [Authorize(Role.admin)]
-        [HttpDelete]
-        [Route("delete")]
-        public async Task<IActionResult> Delete()
+        //[Authorize(Role.admin)]
```

Comment:
Why are we not authorizing the user here? The user seems to delete an entity. Is everyone allowed to delete users in the system?